packages/context/src/binding.ts

```diff
@@ -131,13 +144,32 @@ export class Binding {
     };
   }
 
+  /**
+   * Get nested properties by path
+   * @param value Value of an object
+   * @param path Path to the property. If not present, the root value will be
+   * returned.
+   */
+  static getDeepProperty(value: BoundValue, path?: string) {
+    path = path || '';
+    const props = path.split('.').filter(Boolean);
+    for (const p of props) {
+      value = value[p];
+      if (value === undefined || value === null) {
+        return value;
+      }
+    }
+    return value;
+  }
+
   public readonly key: string;
   public readonly tags: Set<string> = new Set();
   public scope: BindingScope = BindingScope.TRANSIENT;
   public type: BindingType;
+  public options: BindingOptions = {};
```

Comment:
Have you considered moving optional fields like `tags` and `scope` to `options`?

Reply:
No. The newly introduced `options` is mostly for configuration of the bound service. It's different from `tags` and `scope`, which are flags to control how the binding is organized and connected to the context.

Reply:
Fair enough. It would be difficult to move tags and scope to options anyways, since options can be a promise.